packages/date-picker/src/shared/components/DateInput/DateInputSegment/DateInputSegment.tsx

```diff
@@ -123,17 +130,23 @@ export const DateInputSegment = React.forwardRef<
         }
 
         case keyMap.Backspace: {
-          const numChars = value.length;
-
-          // If we've cleared the input with backspace,
-          // fire the custom change event
-          if (numChars === 1) {
-            onChange({
-              segment,
-              value: '',
-              meta: { key },
-            });
+          const preVal = target.value;
+
+          // reset the input on backspace
+          target.value = '';
```

Comment:
I don't know if we should manually be setting `target.value` here. The component is controlled, and the value should be cleared in response to the `onChange` event we're firing

Reply:
Can you elaborate a little more on this?

Reply:
`DateInputSegment` is always a controlled component—i.e. it accepts `value` & `onChange` props. 
If we manually set `target.value` _without_ calling `onChange` internally, then the controlling component won't know value is updated and could have a stale `value` state. 
Even if we _do_ call `onChange` internally, whatever we set `target.value` to will be reset by the incoming `value` prop

Reply:
Chatted in person -- we call `onChange` internally inside `DateInputSegment` and inside `DatePickerInput => handleSegmentChange` we update the value of the input using the up-to-date value from `segmentChangeEvent`